docs/RECIPE_SCHEMA.md

```diff
@@ -0,0 +1,319 @@
+# Cellpack Recipe Schema 1.0
+## Recipe definition properties
+
+### name
+
+_Required string._
+
+Unique recipe name
+
+### version
+
+_Required string._
+
+Version of the recipe
+
+
+#### example of recipe definition:
+
+```JSON
+"recipe": {
+    "version": "1.0",
+    "name": "NM_Analysis_FigureA"
+    },
+```
+----
+
+## Options properties
+
+### cancelDialog
+
+_boolean_
+
+### hackFreepts
+
+### windowsSize
+_number_
+
+### use_gradient
+
+_Optional boolean_
+
+### placeMethod
+
+_Optional enum_. One of `"RAPID"`, `"jitter"`, `"spheresBHT"`, `"pandaBullet"`
+
+Will be use if placeMethod isn't in an ingredient setup
+
+`"RAPID"`
+
+`"jitter"`
+
+`"spheresBHT"`
+
+`"pandaBullet"`
+
+### saveResult
+_boolean_
+
+### runTimeDisplay
+_boolean_
+
+### overwritePlaceMethod
+_boolean_
+
+### innerGridMethod
+_enum_. One of `"bhtree"`, `"jordan"`, `"sdf"`, `"jordan3"`, `"pyray"`, `"floodfill"`, `"binvox"`, `"trimesh"`, `"scanline"`
+ `"bhtree"`
+ 
+ `"jordan"`
+
+`"sdf"`
+
+ `"jordan3"`
+ 
+ `"pyray"`
+ 
+ `"floodfill"`
+ 
+ `"binvox"`
+ 
+ `"trimesh"`
+ 
+ `"scanline"`
+
+### boundingBox 
+_Required [[number, number, number], [number, number, number]]_
+
+An array of two points that define the corners of the bounding box.
+
+### gradients 
+_Optional string[]_
+
+An array that defines the names of directional gradients to use.
+
+### smallestProteinSize
+_Required number_
```

Comment:
@gj210 are these required? (with largest protein size)

Reply:
This is not required, but is a hack to allow the user to manipulate the voxel size of the MasterGrid instead of accepting the calculated defaults... e.g. a sparse grid will lead to very fast but potentially more sparse packing, and allows unexpected things like for an object to jitter beyond its pickedPoint, which is effectively an inefficient way to pack. It can be helpful to force more attempts when packing densely, e.g. odd-shaped lipids into a lipid bilayer, or for sphereTrees that are overly-coarse

Reply:
If deleted, then this value will be calculated correctly by the algorithm, based on the spheretree input file (packing radius) or the packingRadius calculated from ingredients that use primatives for their geometries.